libp2p/transports/quictransport.nim

```diff
@@ -192,19 +187,30 @@ method handle*(m: QuicMuxer): Future[void] {.async: (raises: []).} =
     trace "finished handling stream"
     doAssert(chann.closed, "connection not closed by handler!")
 
-  try:
-    while not m.session.atEof:
-      let stream = await m.session.getStream(Direction.In)
-      asyncSpawn handleStream(stream)
-  except QuicTransportError as exc:
-    trace "Exception in quic handler", msg = exc.msg
+  while not m.session.atEof:
+    let stream =
+      try:
+        await m.session.getStream(Direction.In)
+      except CancelledError:
+        # keep handling, until connection is closed
+        continue
+      except QuicError as exc:
```

Comment:
It feels weird to have the stop condition of a loop to be an exception. Maybe there's a better way of checking if the connection was closed? One that we could use to do 

`while not m.session.atEof and not m.connection.closed:` (pseudocode)